### PR TITLE
Add support for functions and procedures in mysql dialect

### PR DESCRIPTION
**Pull request by silverbullettruck2001**
Added support for the CREATE/DROP PROCEDURE statement for the mysql dialect
Added specific allowed/required syntaxes for CREATE/DROP FUNCTION within the mysql dialect

### CODE REVIEW

**Review by dmateusp — COMMENTED** [review]
Just the little CHANGELOG note, and missing the DEFINER = user case

otherwise it all looks good! thanks for putting the PR together

File: CHANGELOG.md
Comment (silverbullettruck2001) [reply CHANGELOG.md:19]: I added it in a previous commit, but wasn't really sure if it was appropriate for me to declare the minor release at this point.  I can add it back if you think it is appropriate?

File: src/sqlfluff/dialects/dialect_mysql.py
Comment (silverbullettruck2001) [reply src/sqlfluff/dialects/dialect_mysql.py:167]: I actually left it out on purpose.  I spent significant time in trying to get it to parse the DEFINER statement but it wasn't working correctly.  So I left it out so I could open a new issue to address it later since that statement is optional.

File: CHANGELOG.md
Comment (silverbullettruck2001) [reply CHANGELOG.md]: I have added myself as requested.

File: CHANGELOG.md
Comment (dmateusp) [inline CHANGELOG.md:16]: This section was fine here, (under ##Unreleased)

File: CHANGELOG.md
Comment (dmateusp) [inline CHANGELOG.md:13]: Every contributor should appear once, each PR should be separated by a comma (also don't forget the `#`)

File: CHANGELOG.md
Comment (dmateusp) [reply CHANGELOG.md:19]: I think you should leave it to the person making the release to replace the ##Unchanged with the new version being released

File: CHANGELOG.md
Comment (dmateusp) [inline CHANGELOG.md:21]: I see a couple of unrelated changes here, was it a merging issue?

File: src/sqlfluff/dialects/dialect_mysql.py
Comment (dmateusp) [reply src/sqlfluff/dialects/dialect_mysql.py:167]: got it! could we open that issue already and refer to it here?

**Review by dmateusp — COMMENTED** [review]
Just a few CHANGELOG related questions still :)

File: CHANGELOG.md
Comment (silverbullettruck2001) [reply CHANGELOG.md:21]: Yes, this was a merging issue

File: src/sqlfluff/dialects/dialect_mysql.py
Comment (silverbullettruck2001) [reply src/sqlfluff/dialects/dialect_mysql.py:167]: I created an issue for this, but I have already started a bunch of new stuff to expand on the mysql dialect.  Instead of reverting all my changes to resolve the issues with the `CHANGELOG` and this documentation item.  Can we proceed with this how it is and get it merged and I will clean up these items in my next PR?

Comment (dmateusp) [reply src/sqlfluff/dialects/dialect_mysql.py:167]: @silverbullettruck2001 sorry but I'm not in favor of leaving the CHANGELOG in an unclean state for even a few commits because it might cause further issues/confusions on other PRs

blocking the changes now explicitly (since the comments above were marked as resolved)

**Review by dmateusp — CHANGES_REQUESTED** [review]

**Review by dmateusp — APPROVED** [review]

**Comment by dmateusp** [comment]
@silverbullettruck2001 thanks for the contribution and sorry I've had to hold this up for the CHANGELOG 🙇

**Pull request merged**
